src/utils/BatchManager.js

```diff
@@ -151,22 +151,18 @@ class BatchManager {
       if (!renderFn || typeof renderFn !== 'function') {
         // component not registered
         context.statusCode = 404;
-        context.duration = msSince(start);
-        throw notFound(name);
+        return Promise.reject(notFound(name));
       }
 
-      let response = null;
-
-      // render the component!
-      try {
-        context.html = renderFn(context.props);
-      } catch (e) {
-        response = Promise.reject(e);
-      } finally {
-        context.duration = msSince(start);
-      }
-
-      return response;
+      return Promise.resolve(renderFn(context.props));
```

Comment:
since this is already inside a `.then`, the `Promise.resolve()` isn't needed here - you can just `return renderFn(context.props)` directly.